types/wlr_text_input.c

```diff
@@ -0,0 +1,182 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <stddef.h>
+
+#include <wlr/types/wlr_seat.h>
+#include <wlr/types/wlr_text_input.h>
+
+#include <wlr/util/log.h>
+
+#include "text-input-unstable-v3-protocol.h"
+#include "util/signal.h"
```

Comment:
More spacing nit: remove all of the empty lines above here and one of the empty lines below here